mlrun/datastore/targets.py

```diff
@@ -1429,6 +1431,215 @@ def as_df(
+    is_online = True
+    support_spark = False
+    support_storey = True
+
+    def __init__(
+        self,
+        name: str = "",
+        path=None,
+        attributes: typing.Dict[str, str] = None,
+        after_step=None,
+        columns=None,
+        partitioned: bool = False,
+        key_bucketing_number: typing.Optional[int] = None,
+        partition_cols: typing.Optional[typing.List[str]] = None,
+        time_partitioning_granularity: typing.Optional[str] = None,
+        after_state=None,
+        max_events: typing.Optional[int] = None,
+        flush_after_seconds: typing.Optional[int] = None,
+        storage_options: typing.Dict[str, str] = None,
+        db_name: str = None,
+        connection_string: str = None,
+        collection_name: str = None,
+        create_collection: bool = False,
+        override_collection: bool = False,
+    ):
+
+        if not all([db_name, collection_name, connection_string]):
+            attr = {}
+        else:
+
+            from pymongo import MongoClient
+
+            mongodb_client = MongoClient(connection_string)
+            all_dbs = mongodb_client.list_database_names()
+            if db_name not in all_dbs:
+                if create_collection:
+                    pass
+                else:
+                    raise ValueError(f"DataBase named {db_name} is not exist")
+            db = mongodb_client[db_name]
+            all_collections = db.list_collection_names()
+            if collection_name not in all_collections:
+                if create_collection:
+                    collection = db[collection_name]
+                    collection.insert_one({"test": "test"})
+                    collection.delete_one({"test": "test"})
+                else:
+                    raise ValueError(
+                        f"Collection named {collection_name} is not exist in {db_name} database"
+                    )
+            elif override_collection:
+                db[collection_name].delete_many({})
+            attr = {
+                "collection_name": collection_name,
+                "db_name": db_name,
+                "connection_string": connection_string,
+            }
+            path = f"mdb:///{connection_string}///{db_name}///{collection_name}"
+        if attributes:
+            attributes.update(attr)
+        else:
+            attributes = attr
+
+        super().__init__(
+            name,
+            path,
+            attributes,
+            after_step,
+            columns,
+            partitioned,
+            key_bucketing_number,
+            partition_cols,
+            time_partitioning_granularity,
+            max_events=max_events,
+            flush_after_seconds=flush_after_seconds,
+            storage_options=storage_options,
+            after_state=after_state,
+        )
+
+    def add_writer_state(
+        self, graph, after, features, key_columns=None, timestamp_key=None
+    ):
+        warnings.warn(
```

Comment:
we are far from 0.9.0 and if you are adding new code, I think it should be gone from the first place.

Reply:
So to delete it? 
I had because all the sources have this method